remoting/src/main/java/org/apache/rocketmq/remoting/common/RemotingHelper.java

```diff
@@ -53,8 +52,10 @@ public static String exceptionSimpleDesc(final Throwable e) {
     }
 
     public static SocketAddress string2SocketAddress(final String addr) {
-        String[] s = addr.split(":");
-        InetSocketAddress isa = new InetSocketAddress(s[0], Integer.parseInt(s[1]));
+        int split = addr.lastIndexOf(":");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

If the lastIndexOf() search on line 55 does not find the substring and returns -1, the call to substring() on line 56 will throw IndexOutOfBoundsException.